Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -601,7 +601,7 @@ public struct PackageBuilder {
         
         // Create the test modules
         return try testsDirContents.filter(shouldConsiderDirectory).flatMap { dir in
-            return [try createModule(dir, name: dir.basename, isTest: true)]
+            return [try createModule(dir, name: dir.basename, isTest: true)].flatMap { $0 }
```

Comment:
Not entirely fair as there was no need for it to be an array in the first place. See the updated logic.